lib/Runtime/Library/TypedArray.cpp

```diff
@@ -606,15 +606,39 @@ namespace Js
             object;
     };
 
-    BOOL TypedArrayBase::HasProperty(PropertyId propertyId)
+    BOOL TypedArrayBase::HasOwnProperty(PropertyId propertyId)
+    {
+        return HasPropertyEx(propertyId) == Property_Found;
+    }
+
+    inline BOOL TypedArrayBase::CanonicalNumericIndexString(PropertyId propertyId, ScriptContext *scriptContext)
+    {
+        PropertyString *propertyString = scriptContext->GetPropertyString(propertyId);
```

Comment:
Or just avoid the string creation and rewrite the CanonicalNumbericIndexString check to process a string buffer instead.
Anyways, This can be optimized if there is a need later

---
In reply to: [105298966](https://github.com/Microsoft/ChakraCore/pull/2640#discussion_r105298966) [](ancestors = 105298966,105254349)